ontopy/ontology.py

```diff
@@ -283,39 +283,45 @@ def get_by_label(
         The current implementation also supports "*" as a wildcard
         matching any number of characters. This may change in the future.
         """
-        # pylint: disable=too-many-arguments,too-many-branches
+        # pylint: disable=too-many-arguments,too-many-branches,invalid-name
         if not isinstance(label, str):
             raise TypeError(
                 f"Invalid label definition, must be a string: {label!r}"
             )
+
+        # Comment to reviewer:
+        # Do we really want to disallow space in label?
+        # I think we wanted ontopy to work with any ontology, also ontologies
+        # having space in their labels.
+        # Rather than enforcing EMMO conventions here, I think the user should
+        # run emmocheck on the ontology - it will alert about labels with
+        # spaces
         if " " in label:
             raise ValueError(
                 f"Invalid label definition, {label!r} contains spaces."
             )
+
         if self._label_annotations is None:
             for iri in DEFAULT_LABEL_ANNOTATIONS:
                 try:
                     self.add_label_annotation(iri)
                 except ValueError:
                     pass
 
+        # Comment to reviewer:
+        # Since providing the prefix as argument is more explicit, I think
+        # that should take precedence.  Changed it below...
         splitlabel = label.split(":", 1)
-        if len(splitlabel) > 2:
-            raise ValueError(
-                f"Invalid label definition, {label!r}"
-                " contains more than one ':' ."
-                "The string before ':' indicates the prefix. "
-                "The string after ':' indicates the label."
-            )
```

Comment:
I removed it because `len(splitlabel)` is always equal to or less than 2, hence raise will never be called.
This is because of the second argument to `split()` on line 302 ensuring that the label is only splittet once.

But simply removing this check may be the wrong medicine. 

The issue is little tricky. On one hand we don't want to exclude using ontopy with ontologies having colon in their labels. On the other hand, it is difficult to distinguish between a colon indicating a prefix and a colon that is part of the label. 

Should we add a new boolean argument `colon_ok` together with a corresponding ontology-wise setting, about whether to labels with colon are acceptable (in which case label prefixes are turned off). Or is it too much hassle for a corner case?

Reply:
I think that is a good idea.

Reply:
Done